tensorly/decomposition/_tr.py

```diff
@@ -0,0 +1,102 @@
+import tensorly as tl
+from ..tr_tensor import validate_tr_rank, TRTensor
+
+
+def tensor_ring(input_tensor, rank, mode=0, verbose=False):
+    """Tensor Ring decomposition via recursive SVD
+
+        Decomposes `input_tensor` into a sequence of order-3 tensors (factors) [1]_.
+
+    Parameters
+    ----------
+    input_tensor : tensorly.tensor
+    rank : {int, int list}
+            maximum allowable TR rank of the factors
+            if int, then this is the same for all the factors
+            if int list, then rank[k] is the rank of the kth factor
+    mode : int, default is 0
```

Comment:
When would we want to not start by 0? Does it change much the final decomposition?

Reply:
The authors in the original paper state:
> TR-SVD might obtain different representations by choosing a different mode as the start point. This indicates that TR-ranks is not necessary to be the global optimum in TR-SVD.

Generally, the ranks on the sides are smaller than the ones in middle. Changing the mode allows to shift the sides and to explore different ranks.